### PR TITLE
ECS Service - Fix Port Mapping to Host and add daemon placement strategy

### PR DESCRIPTION
**Pull request by roleyfoley**
Fix: for Port Mapping configuration which allows for Host - Container mappings to be defined without a load balancer. 

Feature: ECS placement strategies - initially adds daemon support which ensures that all ECS instance have a single instance of the service running all the time

### CODE REVIEW

File: aws/templates/id/id_ecs.ftl [hunk 1]
```diff
@@ -240,6 +240,18 @@
...
                     "Type" : BOOLEAN_TYPE,
                     "Default" : false
+                },
+                {
+                    "Names" : "Placement",
+                    "Children" : [
+                        {
+                            "Names" : "Strategy",
+                            "Type" : STRING_TYPE,
+                            "Values" : ["daemon"],
```
Comment (ml019) [inline aws/templates/id/id_ecs.ftl]: Need "" as one of the value if it is the default

File: aws/templates/resource/resource_ecs.ftl [hunk 2]
```diff
@@ -159,7 +169,16 @@
                     }
                 },
                 networkMode == "awsvpc"
-            ) + 
+            ) +
+            valueIfTrue(
+                {
+                    "SchedulingStrategy" : "DAEMON"
+                },
+                (placement.Strategy?has_content) && (placement.Strategy == "daemon"),
```
Comment (ml019) [inline aws/templates/resource/resource_ecs.ftl]: Strategy has a default so should just be able to do straight comparison

File: aws/templates/id/id_ecs.ftl (hunk 1 above)
Comment (roleyfoley) [reply aws/templates/id/id_ecs.ftl]: Fixed

File: aws/templates/resource/resource_ecs.ftl (hunk 2 above)
Comment (roleyfoley) [reply aws/templates/resource/resource_ecs.ftl]: Fixed

**Comment by roleyfoley** [comment]
@ml019 fixed up

**Pull request merged**